Default transcript dumping in "statistics dump" to false

qxy11 opened the pull request:
### Summary
Currently, if the setting `interpreter.save-transcript` is enabled, whenever we call "statistics dump", it'll default to reporting a huge list of transcripts which can be a bit noisy. This is because the current check `GetIncludeTranscript` returns `!GetSummaryOnly()` by default if no specific transcript-setting option is given in the statistics dump command (ie. `statistics dump --transcripts=false` or `statistics dump --transcripts=true`). Then when `interpreter.save-transcript` is enabled, this saves a list of transcripts, and the transcript list ends up getting logged by default.

These changes default the option to log transcripts in the `statistics dump` command to "false". This can still be enabled via the `--transcripts` option if users want to see a transcript. Since `interpreter.save-transcript` is false by default, the main delta is that if `interpreter.save-transcript` is true and summary mode is false, we now disable saving the transcript.

This also adds a warning to 'statistics dump --transcript=true' when
interpreter.save-transcript is disabled, which should help users understand
why transcript data is empty.

### Testing

#### Manual testing
Tested with `settings set interpreter.save-transcript true` enabled at startup on a toy hello-world program:
```
(lldb) settings set interpreter.save-transcript true
(lldb) target create "/home/qxy11/hello-world/a.out"
Current executable set to '/home/qxy11/hello-world/a.out' (x86_64).
(lldb) statistics dump
{
  /* no transcript */
}
(lldb) statistics dump --transcript=true
{
"transcript": [
    {
      "command": "statistics dump",
      "commandArguments": "",
      "commandName": "statistics dump",
      "durationInSeconds": 0.0019650000000000002,
      "error": "",
      "output": "{...
    },
    {
      "command": "statistics dump --transcript=true",
      "commandArguments": "--transcript=true",
      "commandName": "statistics dump",
      "timestampInEpochSeconds": 1750720021
    }
  ]
}
```
Without `settings set interpreter.save-transcript true`:
```
(lldb) target create "/home/qxy11/hello-world/a.out"
Current executable set to '/home/qxy11/hello-world/a.out' (x86_64).
(lldb) statistics dump
{
  /* no transcript */
}
(lldb) statistics dump --transcript=true
{
  /* no transcript */
}
warning: transcript requested but none was saved. Enable with 'settings set interpreter.save-transcript true'
```

#### Unit tests
Changed unit tests to account for new expected default behavior to `false`, and added a couple new tests around expected behavior with `--transcript=true`.
```
lldb-dotest -p TestStats ~/llvm-sand/external/llvm-project/lldb/test/API/commands/statistics/basic/
```

qxy11 committed:
Disable "transcript" in "statistics dump" by default

qxy11 committed:
Update some comments

qxy11 commented:
cc @jeffreytan81 @clayborg @dmpots

Michael137 reviewed (approved):
Seems sensible to me. Should we issue a warning if a user runs `statistics dump --transcript=true` but the `interpreter.save-transcript` is not turned on?

qxy11 committed:
Add transcript warning and refactor GetIncludeTranscript()

Summary:
Add warning to 'statistics dump --transcript=true' when
interpreter.save-transcript is disabled, helping users understand
why transcript data is empty.

Also refactor StatisticsOptions::GetIncludeTranscript() to use
value_or(false) instead of the verbose has_value()/value() pattern
to address PR comments.

Michael137 reviewed (approved):
LGTM

Any chance we could add a test for the warning?

qxy11 committed:
Add a unit test around warning msg

qxy11 commented:
Added a unit test around the warning message as well. I'll probably need some help with landing this.

qxy11 committed:
Run lints
